cunumeric/thunk.py

```diff
@@ -179,6 +179,23 @@ def transpose(self, rhs, axes, stacklevel):
         """
         raise NotImplementedError("Implement in derived classes")
 
+    def choose(
+        self,
+        *args,
+        choices,
+        out,
+        is_tuple,
+        mode="rase",
```

Comment:
You no longer need to pass `mode` to the thunk functions, remove from this signature and deferred.py, eager.py, lazy.py